reactioncommerce/reaction
# Support plugins directly registering React components

Impact: **minor**  
Type: **feature**

## Issue
Previously you had to make a Blaze template to wrap a React component in order to register it for "paymentSettings", "taxSettings", "marketplaceMerchantSettings", or "coreOrderShipmentWorkflow" UI extension.

## Solution
Now a plugin can directly register `template: "SomeReactTemplate"` for these UI areas as long as "SomeReactTemplate" matches how it was registered with the first argument of `registerComponent`.

## Breaking changes
None. Registering a Blaze template name still works, too.

## Testing
1. Verify that the affected UI areas still display and function properly with the standard installation.
2. Verify that a React component name can be passed as the `template` option for at least one of these areas. If you alter an existing plugin to test this, you'll also need to drop the Packages collection and restart since packages that already exist aren't updated on startup.
